IsNaN operator (#13369)
openvinotoolkit/openvino: docs/ops/comparison/IsNaN.md

# IsNaN {#openvino_docs_ops_comparison_IsNaN_10}

**Versioned name**: *IsNaN-10*

**Category**: *Comparison*

**Short description**: *IsNaN* returns the boolean mask of a given tensor which maps `NaN` to `True`.

**Detailed description**: *IsNaN* returns the bolean mask of the input tensor in which `True` corresponds to `NaN` and `False` to other values.
* The output tensor has the same shape as input tensor.
* The `i`'th element of the output tensor is `True` if  `i`'th element of the input tensor is `NaN`. Otherwise it is `False`.
* For example, for given input tensor `[NaN, 2.1, 3.7, NaN]` the output tensor is `[True, False, False, True]`.

**Inputs**:

*   **1**: A tensor of type *T* and arbitrary shape. **Required.**

**Outputs**:

> *   **1**: A boolean tensor of the same shape as the input tensor.

Reviewer: I suggest to highlight the `boolean` word
```suggestion
*   **1**: A `boolean` tensor of the same shape as the input tensor.
```

Author: done

**Types**

> * *T*: `bfloat16`, `double`, `float`, `float16`.

Reviewer: Some time ago it was decided to not use specific types in the spec, common approach is:
`any supported floating-point type`. 
Also `_IN` suffix can be added to underline it is only about input types.
```suggestion
* **T_IN**: any supported floating-point type.
```

Author: done

**Example**

```xml

> <layer ... type="IsNaN">
>     <input>
>         <port id="0">
>             <dim>256</dim>
>             <dim>56</dim>
>         </port>
>      </input>
>     <output>
>         <port id="1">

Reviewer: I would recommend to add precision to the examples
```suggestion
<layer ... type="IsNaN">
    <input>
        <port id="0" precision="FP32">
            <dim>256</dim>
            <dim>56</dim>
        </port>
    </input>
    <output>
        <port id="1" precision="BOOL">
```

Author: done

# IsNaN {#openvino_docs_ops_comparison_IsNaN_10}

**Versioned name**: *IsNaN-10*

**Category**: *Comparison*

> **Short description**: *IsNaN* returns boolean mask of a given tensor which mapps `NaN` to `True`.

Reviewer 2: ```suggestion
**Short description**: *IsNaN* returns the boolean mask of a given tensor which maps `NaN` to `True`.
```

Author: Thanks, done :)

**Detailed description**: *IsNaN* returns the bolean mask of the input tensor in which `True` corresponds to `NaN` and `False` to other values.
* The output tensor has the same shape as input tensor.
* The `i`'th element of the output tensor is `True` if  `i`'th element of the input tensor is `NaN`. Otherwise it is `False`.

> * For example, for given input tensor `[NaN, 2.1, 3.7, NaN]` the output tensor is `[True, False, False, True]`.

Reviewer 2: Thanks for the clear example! How is infinity handled?

Author: Infinity is handled the same way as in numpy - it is mapped to False.

Reviewer 2: Thanks, could you add that to the example code or description?

Author: I have added `Inf` case to the example.

**Attributes**

>     No attributes available.

Reviewer: In my opinion such explicit info is useful, as it makes easier and faster to check if the op has attributes or not.
**Attributes** section is currently a part of every op specification, even for those without "attributes".

It should be aligned, **Attributes**: *IsNaN* operation has no attributes.

Example of other ops :
https://github.com/openvinotoolkit/openvino/search?l=Markdown&q=%22has+no+attributes%22&type=code

>     No attributes available.

Author: I decided to follow the convention proposed by @mitruska

**Inputs**:

*   **1**: A tensor of type *T* and arbitrary shape. **Required.**

**Outputs**:

> *   **1**: A tensor of type *T_OUT* and the same shape as input tensor.

Author: done
